### PR TITLE
feat(starlette): added http.route support for root spans in starlette and fastapi

### PR DESCRIPTION
**Pull request by asdftamir**
Added support for the http.route tag in starlette (hence also in fastapi).
The tag is present only in root spans, and not in any sub application spans
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/contrib/starlette/patch.py [hunk 1]
```diff
@@ -138,13 +139,17 @@ def traced_handler(wrapped, instance, args, kwargs):
...
     elif request_spans and resource_paths:
+        route = "".join(resource_paths)
         if scope.get("method"):
-            request_spans[0].resource = "{} {}".format(scope["method"], "".join(resource_paths))
+            request_spans[0].resource = "{} {}".format(scope["method"], route)
         else:
-            request_spans[0].resource = "".join(resource_paths)
-
+            request_spans[0].resource = route
+        request_spans[0].set_tag(http.ROUTE, route)
```
Comment (mabdinur) [inline ddtrace/contrib/starlette/patch.py:152]:
```suggestion
        request_spans[0].set_tag_str(http.ROUTE, route)
```

[the remark this answers is not in the document]
Comment (asdftamir) [reply ddtrace/contrib/starlette/patch.py]: Good point. I checked and from what I've seen ASGI doesn't support route information, and each integration has a different way of getting that info

File: releasenotes/notes/add-http-route-to-starlette-2539a132853213d6.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+other:
```
Comment (mabdinur) [inline releasenotes/notes/add-http-route-to-starlette-2539a132853213d6.yaml:2]:
```suggestion
features:
```

File: releasenotes/notes/add-http-route-to-starlette-2539a132853213d6.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  - |
+    starlette: Add http.route tag to ``starlette.request`` spans.
+  - |
+     fastapi: Add http.route tag to ``fastapi.request`` spans.
```
Comment (mabdinur) [inline releasenotes/notes/add-http-route-to-starlette-2539a132853213d6.yaml:6]:
```suggestion
    starlette: Add http.route tag to ``starlette.request`` spans.
  - |
    fastapi: Add http.route tag to ``fastapi.request`` spans.
```

**Review by mabdinur — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**